Add plugin: Backgroundset

Youngmoss opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin:

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [ ]  macOS
  - [x]  Linux
  - [x]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

Youngmoss commented:
/skip
？github-actions bot say it need main.js,but now need remove?I didn't see any other compiled it

MihailKovachev commented:
You need to upload main.js as part of your release, not commit it to your repo.

https://docs.obsidian.md/Plugins/Releasing/Submit+your+plugin

Youngmoss commented:
sorry,i mix repo and release.it's the first time i pull request.

Youngmoss commented:
It's hard to avoid that in background plugin.Additionally, it's important to ensure that the plugin is not used in conjunction with themes that already have background styles to prevent conflicts
```
  UpdateBackground(doc: Document = activeDocument) {
    document.body.style.setProperty('--obsidian-editor-background-image', `url('${this.settings.imageUrl}')`);
    document.body.style.setProperty('--obsidian-editor-background-opacity', `${this.settings.opacity}`);
    document.body.style.setProperty('--obsidian-editor-background-bluriness', `blur(${this.settings.bluriness})`);
    document.body.style.setProperty('--obsidian-editor-background-input-contrast', this.settings.inputContrast ? '#ffffff17' : 'none');
  }
```
this is Background Image Plugin' code
/skip

Youngmoss commented:
/skip

joethei commented:
Remove the `node_modules` folder from your repo, and add it to your .gitignore.
That folder should never be uploaded to GitHub

["minAppVersion": "0.1.0",](https://github.com/Youngmoss/obsidian-backgroundset/blob/48f4c5b51aba56aa485d190e8b44745c2a8705c9/manifest.json#L3)
That is way to low, community plugins were not a thing back then.

[MySettingTab](https://github.com/Youngmoss/obsidian-backgroundset/blob/48f4c5b51aba56aa485d190e8b44745c2a8705c9/main.ts#L12), [MyPlugin](https://github.com/Youngmoss/obsidian-backgroundset/blob/48f4c5b51aba56aa485d190e8b44745c2a8705c9/main.ts#L143)
Rename these sample class names.

[.setName('Set your Background defaultFolder')](https://github.com/Youngmoss/obsidian-backgroundset/blob/48f4c5b51aba56aa485d190e8b44745c2a8705c9/main.ts#L24), [.setName('Apply random Background')](https://github.com/Youngmoss/obsidian-backgroundset/blob/48f4c5b51aba56aa485d190e8b44745c2a8705c9/main.ts#L98), [.setButtonText('Apply Background')](https://github.com/Youngmoss/obsidian-backgroundset/blob/48f4c5b51aba56aa485d190e8b44745c2a8705c9/main.ts#L101), [.setName('Delete Background')](https://github.com/Youngmoss/obsidian-backgroundset/blob/48f4c5b51aba56aa485d190e8b44745c2a8705c9/main.ts#L109), [.setButtonText('Delete Background')](https://github.com/Youngmoss/obsidian-backgroundset/blob/48f4c5b51aba56aa485d190e8b44745c2a8705c9/main.ts#L112)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI) and make sure that there is a space between each word.

[.setName('Set the change time(min)')](https://github.com/Youngmoss/obsidian-backgroundset/blob/48f4c5b51aba56aa485d190e8b44745c2a8705c9/main.ts#L53), [.setName('Set the opacity')](https://github.com/Youngmoss/obsidian-backgroundset/blob/48f4c5b51aba56aa485d190e8b44745c2a8705c9/main.ts#L67), [.setName('Set the fade out-in time(ms)')](https://github.com/Youngmoss/obsidian-backgroundset/blob/48f4c5b51aba56aa485d190e8b44745c2a8705c9/main.ts#L85)
Remove the "Set the" part of the names, this is a settings tab, everything in there can be set.

Youngmoss commented:
I have completed the following changes:

1. Removed the `node_modules` folder and added it to `.gitignore`.
2. Updated the `minAppVersion` to 0.8.0 to avoid using a very low version number.
3. Renamed the `MySettingTab` and `MyPlugin` class names.
4. Changed UI text to sentence case and ensured a space between each word.
5. Removed the "Set the" part from setting names to make them more concise.
6. Removed `globalThis.imageAdressList` and switched to managing the variable through the plugin instance.
```
this.imageAdressList = [];
delete this.imageAdressList;
```

The changes have been completed.

Youngmoss commented:
/skip 
for second，i said last time
for [1]，it just a text to declear
